orca-queue-redis/src/main/kotlin/com/netflix/spinnaker/orca/q/redis/RedisQueue.kt

```diff
@@ -58,17 +56,19 @@ class RedisQueue(
   private val messagesKey = queueName + ".messages"
   private val attemptsKey = queueName + ".attempts"
   private val locksKey = queueName + ".locks"
+  private val hashKey = queueName + ".hash"
+  private val hashesKey = queueName + ".hashes"
 
   override fun poll(callback: (Message, () -> Unit) -> Unit) {
     pool.resource.use { redis ->
       redis.apply {
-        pop(queueKey, unackedKey, ackTimeout)
-          ?.also { id -> hincrBy(attemptsKey, id, 1) }
-          ?.let { id ->
-            readMessage(id) { payload ->
-              callback.invoke(payload) {
-                ack(id)
-                fire<MessageAcknowledged>()
+        zrangeByScore(queueKey, 0.0, score(), 0, 1)
+          .firstOrNull()
+          ?.takeIf { id -> acquireLock(id) }
+          ?.also { id ->
+            readMessage(id) { message ->
+              callback.invoke(message) {
```

Comment:
this looks to me like the redis connection is held open while the callback handles the message, is that expected?